presto-main/src/test/java/io/prestosql/operator/scalar/TestArrayMatchFunctions.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.prestosql.spi.type.BooleanType;
+import org.testng.annotations.Test;
+
+public class TestArrayMatchFunctions
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testAllMatch()
+    {
+        assertFunction("all_match(ARRAY [5, 7, 9], x -> x % 2 = 1)", BooleanType.BOOLEAN, true);
+        assertFunction("all_match(ARRAY [true, false, true], x -> x)", BooleanType.BOOLEAN, false);
```

Comment:
Add a variant of this that tests what happens if the lambda returns null (function should return null):
```java
assertFunction("all_match(ARRAY [true, false, NULL], x -> x)", BooleanType.BOOLEAN, null);
```
Same for the other functions.